xarray/core/dataset.py

```diff
@@ -862,8 +868,12 @@ def _construct_direct(
         return obj
 
     @classmethod
-    def _from_vars_and_coord_names(cls, variables, coord_names, attrs=None):
-        return cls._construct_direct(variables, coord_names, attrs=attrs)
+    def _from_vars_and_coord_names(
+        cls, variables, coord_names, indexes=None, attrs=None
+    ):
+        return cls._construct_direct(
+            variables, coord_names, indexes=indexes, attrs=attrs
+        )
```

Comment:
I realize this isn't part of this PR, so maybe for another day; does calling `_from_vars_and_coord_names` do anything different from calling `_construct_direct`?